server/index.js

```diff
@@ -40,6 +38,109 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart', (req, res, next) => {
+  const sql = `
+    select *
+    from "carts"
+  `;
+  db.query(sql)
+    .then(result => {
+      if (!req.session.cartId) {
+        res.json([{}]);
+      } else {
+        const sql9000 = `
+        select "c"."cartItemId",
+                "c"."price",
+                "p"."productId",
+                "p"."image",
+                "p"."name",
+                "p"."shortDescription"
+        from "cartItems" as "c"
+        join "products" as "p" using ("productId")
+        where "c"."cartId" = ${req.session.cartId}
+        `;
+        db.query(sql9000)
+          .then(rez => {
+            res.json(rez.rows);
+          });
+      }
+    })
+    .catch(err => next(err));
+});
+
+app.post('/api/cart/:productId', (req, res, next) => {
+  if (parseInt(req.params.productId)) {
+    const sql = `
+      select "price"
+      from "products"
+      where "productId" = ${req.params.productId}
+    `;
+    db.query(sql)
+      .then(result => {
+        if (req.session.cartId) {
+          const sql2 = `
+            select *
+            from "carts"
+            where "cartId" = ${req.session.cartId}
+          `;
+          return db.query(sql2)
+            .then(ress => {
+              const obj = {
+                cartId: ress.rows[0].cartId,
+                price: result.rows[0].price
+              };
+              return (obj);
+            });
```

Comment:
If there is a `req.session.cartId`, we want to avoid querying the database for a `cartId`. We already have it.